src/worker_files.rs

```diff
@@ -19,58 +18,18 @@ async fn image_payload_from_path(
     path: &str,
     img_tfm: &Option<image_processing::ARAwareTransform>,
     encode_images: bool,
+    img_to_rgb8: bool,
 ) -> Result<ImagePayload, image::ImageError> {
     match image_from_path(path).await {
-        Ok(mut new_image) => {
-            let original_height = new_image.height() as usize;
-            let original_width = new_image.width() as usize;
-            let mut channels = new_image.color().channel_count() as i8;
-            let mut bit_depth = (new_image.color().bits_per_pixel()
-                / new_image.color().channel_count() as u16)
-                as usize;
-
-            // Optionally transform the additional image in the same way the main image was
-            if let Some(img_tfm) = img_tfm {
-                new_image = img_tfm.crop_and_resize(&new_image, None).await;
-            }
-
-            let height = new_image.height() as usize;
-            let width = new_image.width() as usize;
-
-            // Encode the image if needed
-            let mut image_bytes: Vec<u8> = Vec::new();
-            if encode_images {
-                if new_image.color() != image::ColorType::Rgb8 {
-                    new_image = image::DynamicImage::ImageRgb8(new_image.to_rgb8());
-                    bit_depth = (new_image.color().bits_per_pixel()
```

Comment:
there was a small bug here though, in that _channels_ was not updated, so RGBA would return 4 channels. Fixed